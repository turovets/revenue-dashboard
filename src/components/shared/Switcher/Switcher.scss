@import 'src/styles/variables';

.Switcher-root {
  width: 100%;
}
// TODO: for some reason library styles renders with bigger priority than component ones.
//  Need to investigate and remove !important

.Switcher-btn {
  color: $common-text-primary !important;

  &:disabled {
    opacity: 0.5;
  }
}

.Switcher-selected {
  border-color: $common-color-primary !important;
  color: $common-color-primary !important;

  &:not(:last-child) {
    border-right-color: initial;
  }
}
