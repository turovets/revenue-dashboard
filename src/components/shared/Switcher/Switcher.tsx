import React, { Dispatch, memo, SetStateAction } from 'react';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Button from '@material-ui/core/Button';

import './Switcher.scss';

type SwitcherItem = string;
type SwitcherProps = {
  disabled?: boolean;
  activeItem: SwitcherItem;
  items: string[];
  // TODO: make switcher generic to pass data type
  onChangeHandler: Dispatch<SetStateAction<any>>;
};

const Switcher = ({
  disabled,
  items,
  onChangeHandler,
  activeItem,
}: SwitcherProps) => {
  const onBtnClick = (item: SwitcherItem) => () => {
    onChangeHandler(item);
  };

  return (
    <ButtonGroup className="Switcher-root" disabled={disabled}>
      {
        items.map((item) => (
          <Button
            key={item}
            className={`Switcher-btn ${activeItem === item ? 'Switcher-selected' : ''}`}
            onClick={onBtnClick(item)}
          >
            {item}
          </Button>
        ))
      }
    </ButtonGroup>
  );
};

export default memo(Switcher)     ;
