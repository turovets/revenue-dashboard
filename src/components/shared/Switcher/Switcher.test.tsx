import React from 'react';

import { render } from '@testing-library/react';

import Switcher from './Switcher';

describe('Switcher', () => {
  it('should render component correctly', () => {
    const props = {
      activeItem: 'Monthly',
      items: ['Monthly', 'Weekly'],
      onChangeHandler: jest.fn(),
    };
    const { asFragment } = render(
      <Switcher {...props} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
