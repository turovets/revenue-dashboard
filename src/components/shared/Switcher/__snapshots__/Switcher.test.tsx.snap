// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiButtonGroup-root Switcher-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined Switcher-btn Switcher-selected"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Monthly
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined Switcher-btn "
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Weekly
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
